sequencer/src/state.rs

```diff
@@ -271,6 +292,9 @@ fn validate_and_apply_header(
         tracing::warn!("Invalid Builder: {}", e);
         return Err(BlockError::InvalidBlockHeader);
     };
+
+    delta.fees_delta.insert(proposed_header.fee_info.account);
```

Comment:
Nit: for deposits in `validate_and_apply_proposal`, we update `fees_delta` in the callee, but here for `validate_and_charge_builder` it's left to the caller. Doesn't really matter but I think it's less confusing if we are consistent here.